analyses/methylation-preprocessing/scripts/01-preprocess-illumina-arrays.R

```diff
@@ -48,6 +50,13 @@ base_dir <- opt$base_dir
 controls_present <- opt$controls_present
 snp_filter <- opt$snp_filter
 
+# read manifest file_name and get Bioassay_ID columns
+man_df <- read.table(file = opt$manifest_file, sep = '\t', header = TRUE)
+man_df <- man_df %>% select(all_of(c("file_name", "Bioassay_ID"))) %>%
+  filter(!grepl("_Red.", file_name))
+man_df$file_name <- gsub("_Grn.idat", "", man_df$file_name)
+man_df$file_name <- gsub(".gz", "", man_df$file_name)
```

Comment:
```suggestion
man_df <- read_tsv(file = opt$manifest_file) %>% 
  select(file_name, Bioassay_ID) %>%
  dplyr::mutate(file_name = gsub("(_Red|_Grn).*", "", file_name)) %>%
  unique()
```

This should get you the same result using tidyverse and be more flexible for any other potential compressions :)